libkodimote/imagecache.cpp

```diff
@@ -173,6 +174,12 @@ void KodiImageCache::imageFetched()
         }
     }
 
+    foreach (const ImageFetchJob::Callback callback, job->callbacks()) {
+        QMetaObject::invokeMethod(callback.object().data(), callback.method().toLatin1(), Qt::QueuedConnection, Q_ARG(int, job->id()));
+    }
+
+    job->deleteLater();
```

Comment:
the networkrequest doesn't seem to take ownership of the originatingObject. So I think this delete is required indeed.

Also I think if a request fails, we should inform the requester of that instead of keeping it waiting forever. Also if a request fails, we should remember that and not try over and over again. Chances are *really* low that it'll change, and those times we'll still catch when the app starts the next time. So better just telling the ui there's nothing to come and it should display the default artwork.